File: 35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python3
#
# [35] Search Insert Position
#
from typing import *
# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)
        while left < right:
            middle = (left+right) // 2
            if nums[middle] < target:
                left = middle + 1
            else:
                right = middle
        return left
if __name__ == "__main__":
    print("\tleft\tright\tmiddle")
    print("for zero:")
    # print(Solution().searchInsert([1,3,5,6],0))
    print("for two:")
    print(Solution().searchInsert([1,3,5,6],5))
# @lc code=end

File: golang/35.search-insert-position.go
/*
 * @lc app=leetcode id=35 lang=golang
 *
 * [35] Search Insert Position
 */
package main

// @lc code=start
func searchInsert(nums []int, target int) int {
	left := 0
	right := len(nums) // 1
	for left < right { // 2
		middle := (left + right) / 2 //
		if nums[middle] < target {   // 4
			left = middle + 1 //5
		} else {
			right = middle
		}
	}
	return left // 6
}

// @lc code=end
